rclcpp/include/rclcpp/executor.hpp

```diff
@@ -361,14 +361,14 @@ class Executor
         for (auto & weak_subscription : group->subscription_ptrs_) {
           auto subscription = weak_subscription.lock();
           if (subscription) {
-            subs.push_back(subscription);
+            memory_strategy_->subs.push_back(subscription);
```

Comment:
I think the `MemoryStrategy` API would look clear if, instead of exposing the `std::vector` directly, provide `add_subscription`, `add_service`, etc. and hide the underlying storage.